### ISSUE TITLE
bundler 1.13.0 seems to break rails new

### ISSUE THREAD

**Issue by jrochkind** [rails/rails#26472]
While I imagine this is a bundler bug not a Rails bug, I can only reproduce it with Rails and haven't figured out what's going on, so am reporting it here. 

~~~
$ ruby -v
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]
$ rails -v
Rails 5.0.0.1

$ rails new testapp
      create
      create  README.md
      create  Rakefile
      create  config.ru
      create  .gitignore
      create  Gemfile
      create  app
      create  app/assets/config/manifest.js
      create  app/assets/javascripts/application.js
      create  app/assets/javascripts/cable.js
      create  app/assets/stylesheets/application.css
      create  app/channels/application_cable/channel.rb
      create  app/channels/application_cable/connection.rb
      create  app/controllers/application_controller.rb
      create  app/helpers/application_helper.rb
      create  app/jobs/application_job.rb
      create  app/mailers/application_mailer.rb
      create  app/models/application_record.rb
      create  app/views/layouts/application.html.erb
      create  app/views/layouts/mailer.html.erb
      create  app/views/layouts/mailer.text.erb
      create  app/assets/images/.keep
      create  app/assets/javascripts/channels
      create  app/assets/javascripts/channels/.keep
      create  app/controllers/concerns/.keep
      create  app/models/concerns/.keep
      create  bin
      create  bin/bundle
      create  bin/rails
      create  bin/rake
      create  bin/setup
      create  bin/update
      create  config
      create  config/routes.rb
      create  config/application.rb
      create  config/environment.rb
      create  config/secrets.yml
      create  config/cable.yml
      create  config/puma.rb
      create  config/spring.rb
      create  config/environments
      create  config/environments/development.rb
      create  config/environments/production.rb
      create  config/environments/test.rb
      create  config/initializers
      create  config/initializers/application_controller_renderer.rb
      create  config/initializers/assets.rb
      create  config/initializers/backtrace_silencers.rb
      create  config/initializers/cookies_serializer.rb
      create  config/initializers/cors.rb
      create  config/initializers/filter_parameter_logging.rb
      create  config/initializers/inflections.rb
      create  config/initializers/mime_types.rb
      create  config/initializers/new_framework_defaults.rb
      create  config/initializers/session_store.rb
      create  config/initializers/wrap_parameters.rb
      create  config/locales
      create  config/locales/en.yml
      create  config/boot.rb
      create  config/database.yml
      create  db
      create  db/seeds.rb
      create  lib
      create  lib/tasks
      create  lib/tasks/.keep
      create  lib/assets
      create  lib/assets/.keep
      create  log
      create  log/.keep
      create  public
      create  public/404.html
      create  public/422.html
      create  public/500.html
      create  public/apple-touch-icon-precomposed.png
      create  public/apple-touch-icon.png
      create  public/favicon.ico
      create  public/robots.txt
      create  test/fixtures
      create  test/fixtures/.keep
      create  test/fixtures/files
      create  test/fixtures/files/.keep
      create  test/controllers
      create  test/controllers/.keep
      create  test/mailers
      create  test/mailers/.keep
      create  test/models
      create  test/models/.keep
      create  test/helpers
      create  test/helpers/.keep
      create  test/integration
      create  test/integration/.keep
      create  test/test_helper.rb
      create  tmp
      create  tmp/.keep
      create  tmp/cache
      create  tmp/cache/assets
      create  vendor/assets/javascripts
      create  vendor/assets/javascripts/.keep
      create  vendor/assets/stylesheets
      create  vendor/assets/stylesheets/.keep
      remove  config/initializers/cors.rb
         run  bundle install
/Users/jrochkind/.gem/ruby/2.3.1/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:23:in `source': uninitialized constant Gem::Source (NameError)
Did you mean?  Gem::Source
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:65:in `extension_dir'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/basic_specification.rb:46:in `gem_build_complete_path'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:2170:in `missing_extensions?'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/basic_specification.rb:71:in `contains_requirable_file?'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `block in find_in_unresolved'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `each'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `find_all'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `find_in_unresolved'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:23:in `source'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/bundler-1.13.0/lib/bundler/rubygems_ext.rb:65:in `extension_dir'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/basic_specification.rb:46:in `gem_build_complete_path'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:2170:in `missing_extensions?'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/basic_specification.rb:71:in `contains_requirable_file?'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `block in find_in_unresolved'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `each'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `find_all'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1067:in `find_in_unresolved'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/bundler-1.13.0/lib/bundler/rubygems_integration.rb:2:in `<top (required)>'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/bundler-1.13.0/lib/bundler.rb:11:in `<top (required)>'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:127:in `require'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:127:in `rescue in require'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:40:in `require'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/railties-5.0.0.1/lib/rails/generators/app_base.rb:373:in `bundle_command'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/railties-5.0.0.1/lib/rails/generators/app_base.rb:401:in `run_bundle'
	from (eval):1:in `run_bundle'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/invocation.rb:133:in `block in invoke_all'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/invocation.rb:133:in `each'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/invocation.rb:133:in `map'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/invocation.rb:133:in `invoke_all'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/group.rb:232:in `dispatch'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/railties-5.0.0.1/lib/rails/commands/application.rb:17:in `<top (required)>'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/railties-5.0.0.1/lib/rails/cli.rb:14:in `<top (required)>'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /opt/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
	from /Users/jrochkind/.gem/ruby/2.3.1/gems/railties-5.0.0.1/exe/rails:9:in `<top (required)>'
	from /Users/jrochkind/.gem/ruby/2.3.1/bin/rails:23:in `load'
	from /Users/jrochkind/.gem/ruby/2.3.1/bin/rails:23:in `<main>'
~~~

What I expect to happen is NOT getting an uncaught exception backtrace as a result of running `rails new`.

**Comment by maclover7** [comment]
Closing in favor of bundler/bundler#4970, since this seems to be a Bundler specific issue. Thank you for reporting!

**Comment by jrochkind** [comment]
Okay, if bundler closes and says it's a Rails bug... what do you want me to do? Cause that's usually how these things go.

**Comment by rafaelfranca** [comment]
It is not a Rails bug.

**Comment by arthurnn** [comment]
@jrochkind [they fixed](https://github.com/bundler/bundler/issues/4970#issuecomment-246491593) and will be released in bundler 1.13.1
